src/chaoticIteration.mli

```diff
@@ -128,7 +128,7 @@ module Make
     (G : G)
     (D : Data with type edge = G.E.t) :
 sig
-  module M : Map.S with type key = G.V.t
+  module M : Map.S with type key = G.V.t and type 'a t = 'a Map.Make(G.V).t
```

Comment:
Is the `type key = G.V.t` constraint still needed if the second one is provided ?

Reply:
Yes. The type `'a t` is an abstract type which does not even have a notion of keys. If you don't supply the equality on on `key`, you would only be able to call the functions of `Map.S` which do not refer to `key`.

Reply:
Fair enough, I read too quickly and am not really fluent with this part of the language. I still don't like the hard-coded dependency to `Map`, but since it's here anyway, I agree to make it fully explicit.